validator/project_validator.go

```diff
@@ -548,17 +548,21 @@ func checkProjectFields(project *model.Project) ValidationErrors {
 
 // Ensures that:
 // 1. a referenced task within a buildvariant task object exists in
-// the set of project tasks
+// the set of project tasks and is not referenced in the task group of the buildvariant
 // 2. any referenced distro exists within the current setting's distro directory
 func ensureReferentialIntegrity(project *model.Project, distroIDs []string, distroAliases []string) ValidationErrors {
 	errs := ValidationErrors{}
 	// create a set of all the task names
 	allTaskNames := map[string]bool{}
+	taskGroupTaskSet := map[string]bool{}
 	for _, task := range project.Tasks {
 		allTaskNames[task.Name] = true
 	}
 	for _, taskGroup := range project.TaskGroups {
 		allTaskNames[taskGroup.Name] = true
+		for _, task := range taskGroup.Tasks {
+			taskGroupTaskSet[task] = true
```

Comment:
Instead of doing map[taskName] -> true, we can do map[taskName] -> taskGroupName; that way at 598 we can print the task group name also.